infor-design/enterprise
# 2989 - Improve handling of Selectable items in Popupmenus

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes several issues described inn #2989, as well as some additional cleanup of the Popupmenu's selection system:
- Submenus on Flex Toolbar-generated action button menus will no longer prematurely close the entire menu when hovered.
- Popupmenus that allow selection on all items now have the correct formatting for list items that properly display the checkbox to their left side (or right on an RTL-based locale)
- Reworked the display of selectable items in general to better support multiple, selectable groups inside popupmenus.  This feature had been broken at some point in the recent past, but works again in this PR.

**Related github/jira issue (required)**:
Closes #2989

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the app.  Then test the following items:

_Proper display of checkmarks:_
- Open http://localhost:4000/components/toolbar-flex/test-more-actions-selectable.html
- Open the more actions menu on the Flex Toolbar in the page.
- Notice the list items are all properly aligned to account for the checkmark on the selected item.  This should have happened automatically (the test doesn't define `is-selectable` by default).

_Hovering submenu items:_
- Open http://localhost:4000/components/toolbar-flex/test-more-actions-selectable.html
- Open the more actions menu on the Flex Toolbar in the page.
- Hover over "More Three" which is a submenu
- The submenu should properly display, instead of "closing" the menu prematurely.

_Testing multiple selectable groups:_
- Open http://localhost:4000/components/popupmenu/example-partially-selectable?theme=uplift
- Click the "Show Partially Checkable Menu" button to open the menu
- If you click on any items labeled "Non-selectable", the items should do nothing but close the menu.
- If you click on any items labeled "Single Selectable", it should select just one checkbox within its "section" and close the menu.  Re-open the menu to confirm the selection changed.
- If you click on any items labeled "Multi Selectable", it should select/deselect the item that was clicked, and leave any of the other checkmarks in the group unchanged. It should also keep the Popupmenu open until something else is clicked.
- This should also work on the submenu.

**Included in this Pull Request**:

**Additional Context**:
- This issue doesn't need to get into 4.23.x unless it's totally ready.  It's scheduled for 4.24.x and should be thoroughly tested before we merge it since it changes some core functionality in Popupmenu (none of which is breaking, however).

----
paging @pwpatton @Sovia
